contracts/deploy-ready/ERC721MinterPauser.sol

```diff
@@ -28,27 +28,32 @@ contract ERC721MinterPauser is Context, AccessControl, ERC721Burnable, ERC721Pau
      * @dev Grants `DEFAULT_ADMIN_ROLE`, `MINTER_ROLE` and `PAUSER_ROLE` to the
      * account that deploys the contract.
      *
-     * See {ERC721-constructor}.
+     * You can specify a `baseURI` to add it as a prefix to all token URIs (and
+     * potentially autogenerate those, leading to large gas savings), or leave
+     * it empty for individual token URIs. See {ERC721-tokenURI}.
      */
-    constructor(string memory name, string memory symbol) public ERC721(name, symbol) {
+    constructor(string memory name, string memory symbol, string memory baseURI) public ERC721(name, symbol) {
         _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
 
         _setupRole(MINTER_ROLE, _msgSender());
         _setupRole(PAUSER_ROLE, _msgSender());
+
+        _setBaseURI(baseURI);
     }
 
     /**
-     * @dev Creates the `tokenId` tokens for `to`.
+     * @dev Creates the `tokenId` tokens for `to`, assigning it `tokenURI`.
      *
-     * See {ERC721-_mint}.
+     * See {ERC721-_mint} and {ERC721-_setTokenURI}.
      *
      * Requirements:
      *
      * - the caller must have the `MINTER_ROLE`.
      */
-    function mint(address to, uint256 tokenId) public {
+    function mint(address to, uint256 tokenId, string memory tokenURI) public {
```

Comment:
I don't like this one. I think it's important to keep the simple `mint`. We can consider adding a separate `mint` with a `tokenURI` argument, but I don't think it's necessary.

Reply:
Shouldn't we then make it mandatory to use a `baseURI`?

Reply:
Not necessarily. Having a `baseURI` argument in the constructor is enough. If someone ones to use an empty string it's their choice to do so.

Reply:
For deploy ready ERC721 my assumption is that most users will want automatic token URI and will use sequential tokenIDs. I would recommend being opinionated and not having a tokenID parameter in the mint function, so that sequential tokenIDs are generated. Non-sequential tokenIDs is then the exception. 

I think most users will just use the address parameter of mint but having tokenURI gives lots of flexibility.

Reply:
It's an interesting idea, but it worries me that it's a bit untested. Also, what if people want the incremental token id feature but not the other things brought in by the deploy ready version? Anyway, I'm just being skeptical but this may actually be a good idea.

Reply:
OpenSea's example uses an incremental token id with a `mint` only accepting an address, which is where I first saw the idea of automatic token URI.
https://github.com/ProjectOpenSea/opensea-creatures/blob/master/contracts/TradeableERC721Token.sol#L31

Reply:
I think it makes sense to leave this one as simple as possible. Thanks!